apps/web/src/app/[locale]/challenge/[id]/solutions/_components/delete/index.tsx

```diff
@@ -0,0 +1,105 @@
+import type { DialogTriggerProps } from "@radix-ui/react-dialog";
+import { useState } from "react";
+import { Dialog, DialogContent, DialogTrigger } from '@repo/ui/components/dialog';
+import { TypographyLarge } from "@repo/ui/components/typography/large";
+import type { ChallengeSolution } from '~/app/[locale]/challenge/[id]/solutions/[solutionId]/page';
+import { Tooltip, TooltipContent, TooltipTrigger } from "@repo/ui/components/tooltip";
+import { getRelativeTime } from "~/utils/relativeTime";
+import { Button } from "@repo/ui/components/button";
+import { TypographyP } from "@repo/ui/components/paragraph";
+import { Markdown } from "@repo/ui/components/markdown";
+import { Label } from "@repo/ui/components/label";
+import { deleteSolution } from "../_actions";
+import { toast } from "@repo/ui/components/use-toast";
+import { useRouter } from "next/navigation";
+
+interface SolutionDeleteDialogProps extends DialogTriggerProps {
+    solution: ChallengeSolution
+}
+
+export function SolutionDeleteDialog({
+    children,
+    solution,
+    ...props
+}: SolutionDeleteDialogProps) {
+    const [isOpen, setIsOpen] = useState(false);
+    const router = useRouter();
+
+    async function handleDeleteSolution() {
+        
+        try {
+            await deleteSolution(solution.id);
+            toast({
+                title: 'Solution Deleted',
+                variant: 'success',
+                description: 'The solution was successfully deleted.',
+            });
+            router.back();
```

Comment:
Hmmm why do we call this?

Reply:
because the route doesn't change even if the solution is deleted, so it results in this 
![image](https://github.com/typehero/typehero/assets/69889382/20af5b2e-756c-470d-90dc-321ce12c456f)